arangod/Aql/AqlValue.cpp

```diff
@@ -1093,10 +1276,47 @@ int AqlValue::Compare(velocypack::Options const* options, AqlValue const& left,
     }
     // fall-through to other types intentional
   }
-
   // if we get here, types are equal or can be treated as being equal
 
   switch (leftType) {
+    case VPACK_INLINE_INT48:
+    case VPACK_INLINE_UINT48:
+    case VPACK_INLINE_INT64:
+    case VPACK_INLINE_UINT64:
+    // if right value type is aslo optimized inline we can optimize comparison
+      if (leftType == rightType) {
+        if (leftType == VPACK_INLINE_UINT64) {
+          uint64_t l = static_cast<uint64_t>(left.toInt64());
+          uint64_t r = static_cast<uint64_t>(right.toInt64());
+          if (l == r) {
+            return 0;
+          }
+          return (l < r ? -1 : 1);
+        } else {
+          int64_t l = left.toInt64();
+          int64_t r = right.toInt64();
+          if (l == r) {
+            return 0;
+          }
+          return (l < r ? -1 : 1);
+        }
+        // intentional fallthrough to double comparison
+      }
+    [[fallthrough]];
+    case VPACK_INLINE_DOUBLE:
+    // here we could only compare doubles in case of right is also inlined
+    // the same is done in VelocyPackHelper::compare for numbers. Equal types are compared
+    // directly - unequal (or doubles) as doubles
```

Comment:
That seems to indicate that the code in VelocyPackHelper::compare is broken. Perhaps we should seize the opportunity to look at that again (perhaps in a different PR), and fix the comparison for large integral values.